addon_service/tests/_factories.py

```diff
@@ -89,6 +99,27 @@ def _create(
         )
 
 
+class ExternalStorageOAuth2ServiceFactory(ExternalStorageServiceFactory):
+    oauth2_client_config = factory.SubFactory(OAuth2ClientConfigFactory)
+
+
+class ExternalStorageOAuth1ServiceFactory(ExternalStorageServiceFactory):
+    oauth1_client_config = factory.SubFactory(OAuth1ClientConfigFactory)
+
+    @classmethod
+    def _create(
+        cls,
+        model_class,
+        credentials_format=CredentialsFormats.OAUTH1A,
+        service_type=ServiceTypes.PUBLIC,
+        *args,
+        **kwargs,
+    ):
+        return super()._create(
+            model_class, credentials_format, service_type, *args, **kwargs
+        )
```

Comment:
We never shoul had `oauth_client_config` as a sub-factory, considering it isnt' relevant and shouldn't be set for non-oauth(2) services.

Would recommend instead just updating the existing `ExternalStorageServiceFactory._create` function to
1) Accept `oauth1_client_config` and `oauth2_client_config` as parameters
2) Use the `credentials_format` to construct the appropriate form of client config using the appropriate factory if one wasn't passed
3) Ignore or error any irrelevant `client_config` that was passed.

Reply:
Note, this is obviously simpler if we decide to collapse down to one `OAuthConfig`.